tracing-mock/src/span.rs

```diff
@@ -59,6 +302,101 @@ impl ExpectedSpan {
         }
     }
 
+    /// Configures this `ExpectedSpan` to expect an explicit parent
+    /// span or to be an explicit root.
+    ///
+    /// **Note**: This method returns a [`NewSpan`] and as such, this
+    /// expectation can only be validated when expecting a new span via
+    /// [`MockCollector::new_span`]. It cannot be validated on
+    /// [`MockCollector::enter`], [`MockCollector::exit`], or any other
+    /// method on [`MockCollector`] that takes an `ExpectedSpan`.
+    ///
+    /// An _explicit_ parent span is one passed to the `span!` macro in the
+    /// `parent:` field.
+    ///
+    /// If `Some("parent_name")` is passed to `with_explicit_parent` then
+    /// the provided string is the name of the parent span to expect.
+    ///
+    /// To expect that a span is recorded with `parent: None`, `None`
+    /// can be passed to `with_explicit_parent` instead.
+    ///
+    /// If a span is recorded without an explicit parent, or if the
+    /// explicit parent has a different name, this expectation will
+    /// fail.
+    ///
+    /// # Examples
+    ///
+    /// The explicit parent is matched by name:
+    ///
+    /// ```
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let span = expect::span()
+    ///     .with_explicit_parent(Some("parent_span"));
+    ///
+    /// let (collector, handle) = collector::mock()
+    ///     .new_span(expect::span().named("parent_span"))
+    ///     .new_span(span)
+    ///     .run_with_handle();
+    ///
+    /// tracing::collect::with_default(collector, || {
+    ///     let parent = tracing::info_span!("parent_span");
+    ///     tracing::info_span!(parent: parent.id(), "span");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    ///
+    /// In the following example, we expect that the matched span is
+    /// an explicit root:
+    ///
+    /// ```
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let span = expect::span()
+    ///     .with_explicit_parent(None);
+    ///
+    /// let (collector, handle) = collector::mock()
+    ///     .new_span(span)
+    ///     .run_with_handle();
+    ///
+    /// tracing::collect::with_default(collector, || {
+    ///     tracing::info_span!(parent: None, "span");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    ///
+    /// In the example below, the expectation fails because the
+    /// span is contextually (rather than explicitly) within the span
+    /// `parent_span`:
```

Comment:
```suggestion
    /// In the example below, the expectation fails because the
    /// span is *contextually*—as opposed to explicitly—within the span
    /// `parent_span`:
```